### DISCUSSION TITLE
Enhance read_parquet function to handle no initial data to allow creating views before data is added.

### DISCUSSION THREAD

**Discussion by K377U** [duckdb/duckdb#11505] [Ideas]
I have hive partitioned parquet data and Im creating views to each table to access the data:

```
CREATE OR REPLACE VIEW {table_name} AS
SELECT {fields}
FROM read_parquet(
['path/*/*.parquet'],
HIVE_PARTITIONING=true,
{hive_types}
)
```

As long as there is data, this works fine.

But when there is no data yet, usually when application is initialized this fails with:

```IO Error: No files found that match the pattern "path/*/*.parquet"```

Would it be possible to allow read_parquet to find zero files? Or can I modify the SQL statement somehow so that it does not fail?

**edit**

It would be enough for my usecase to let `read_parquet` return 0 rows when there is no files after view is defined.

**Comment by Tishj** [comment]
The problem with doing this is that we require at least one file to determine the schema of the view
Which I don't think is possible in this scenario?

**Reply by K377U** [reply]
Im setting all the fields and types for them at `SELECT {fields}` so that should be enough to define the VIEW right? If I would use * then this cant work.

What is nice with this way of defining the table is that I can modify the underlying parquet files as much as I like and things seems to just work.

I wonder if I could just add dummy data and then define the view and remove the dummy after. It would be nice that the query to the view in this case would return 0 rows, but most likely it will fail. That does not matter that much as atleast the initialization is done.

**Reply by K377U** [reply]
Also we could use _metadata.parquet at the table root to define the schema so that in case of no files found and star notation used the view could be created.

**Reply by K377U** [reply]
Tested removing the files and then doing query against the view. View is defined but acccessing the data raises same error. Atleast in this case it would make sense to return 0 rows from `read_parquet` as the view is already defined?

**Comment by djouallah** [comment]
find a workaround, load some dummy data initially just that the folder is not empty
```

**Reply by K377U** [reply]
This works for creating the views. But when querying empty view error is raised.

**Comment by djouallah** [comment]
Add A dummy parquet file

**Reply by K377U** [reply]
Then there is dummy data included in real data. This dummy data must be removed before using the table. And when there is no files the query using the view fails.

**Comment by djouallah** [comment]
I  was in the same situation I add dummy data with blank row, then I run the query and filter it out, yes it is ugly but fixed my issue

**Comment by KrishnaPG** [comment]
Was the `Create view` not supposed to be lazy? Or can be it made to create the view lazily when no files at the time of creation? It can create the view later (lazily) the first time a query is run.
